### PR TITLE
refactor(Scripts/BlackMorass): Refactor black morass bosses to use scheduler instead of event map

### PR DESCRIPTION
**Pull request by ChrisCGalbraith**
## Changes Proposed:
-  Removed events and replaced with TaskScheduler
-  Reviewed timers according to sniffs from Gultask (Currently need clarification)

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14885

## Tests Performed:

- Bosses function and builds without errors

## How to Test the Changes:

1. Go to Black Morass
2. Fight ze bosses

## Known Issues and TODO List:

- [ Needs timers clarified ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by ChrisCGalbraith** [comment]
Would very much appreciate a quick review at this juncture, I am still to get some clarification from @Gultask about some of the timers listed in the issue this addresses.

**Comment by kissingers** [comment]
.Schedule(30s, 35, [this](TaskContext context)

File: src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/boss_aeonus.cpp [hunk 1]
```diff
@@ -95,7 +94,7 @@ struct boss_aeonus : public BossAI
             DoCastVictim(SPELL_SAND_BREATH);
             context.Repeat(10s, 30s);
         })
-        .Schedule(15s, 30s, [this](TaskContext context)
+        .Schedule(1ms, [this](TaskContext context)
```
Comment (Nyeriah) [inline src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/boss_aeonus.cpp:97]: 1ms? is this correct?

Comment (ChrisCGalbraith) [reply src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/boss_aeonus.cpp:97]: The information I was working off said "Immediately". The shortest time I could see previously used somewhere else in the core was 1ms so I used that - do we have a way to schedule something immediately without specifying a time?

**Comment by Nyeriah** [comment]
Sorry, there was a pressing issue I had to address so I took care of the conversions, the timer updates are still valid though

**Comment by ChrisCGalbraith** [comment]
> 

No worries at all!

File: src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/boss_aeonus.cpp (hunk 1 above)
Comment (Nyeriah) [reply src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/boss_aeonus.cpp:97]: There is however I’ve never seen it being used. TaskScheduler::Async

**Comment by Kitzunu** [comment]
Abandoned by author, so closing. Open a new PR or reopen this to continue work

**Pull request closed without merging**